cmd/restic/cmd_check.go

```diff
@@ -287,17 +312,38 @@ func runCheck(opts CheckOptions, gopts GlobalOptions, args []string) error {
 		doReadData(selectPacksByBucket(chkr.GetPacks(), 1, 1))
 	case opts.ReadDataSubset != "":
 		var packs map[restic.ID]int64
-		dataSubset, err := stringToIntSlice(opts.ReadDataSubset)
+		readDataSubset, err := determineReadDataSubsetType(opts.ReadDataSubset)
 		if err == nil {
-			bucket := dataSubset[0]
-			totalBuckets := dataSubset[1]
-			packs = selectPacksByBucket(chkr.GetPacks(), bucket, totalBuckets)
-			packCount := uint64(len(packs))
-			Verbosef("read group #%d of %d data packs (out of total %d packs in %d groups)\n", bucket, packCount, chkr.CountPacks(), totalBuckets)
-		} else {
-			percentage, _ := parsePercentage(opts.ReadDataSubset)
-			packs = selectRandomPacksByPercentage(chkr.GetPacks(), percentage)
-			Verbosef("read %.1f%% of data packs\n", percentage)
+			switch readDataSubset {
+			case Percentage:
+				percentage, err := parsePercentage(opts.ReadDataSubset)
+				if err == nil {
+					packs = selectRandomPacksByPercentage(chkr.GetPacks(), percentage)
+					Verbosef("read %.1f%% of data packs\n", percentage)
+				}
+			case FileSize:
+				repoSize, err := dirSize(gopts.Repo)
+				if err != nil {
+					errors.Fatal("Could not get the file size of repo")
```

Comment:
This line is currently a no-op. The error _must_ be returned to have any effect on the control flow. The also applies to the default case.